lib/utils/buildHelper.js

```diff
@@ -0,0 +1,214 @@
+			if (!dependencyInfo[dep.metadata.name]) {
+				dependencyInfo[dep.metadata.name] = _getTransitiveDependencies(dep, []);
+				_processDependencies(dep);
+			}
+		});
+	}
+
+	_processDependencies(tree);
+	return dependencyInfo;
+}
+
+/**
+ * Creates dependency lists for 'includedDependencies' and 'excludedDependencies'. Regular expressions are directly
+ * applied to a list of all project dependencies so that they don't need to be evaluated in later processing steps.
+ * Generally, includes are handled with a higher priority than excludes. Additionally, operations for processing
+ * transitive dependencies are handled with a lower priority than explicitly mentioned dependencies. The default
+ * dependencies set in the build settings are appended in the end.
+ *
+ * The priority of the various dependency lists is applied in the following order:
+ * <ol>
+ *   <li>includeDependency, includeDependencyRegExp</li>
+ *   <li>excludeDependency, excludeDependencyRegExp</li>
+ *   <li>includeDependencyTree</li>
+ *   <li>excludeDependencyTree</li>
+ *   <li>defaultIncludeDependency, defaultIncludeDependencyRegExp, defaultIncludeDependencyTree</li>
+ * </ol>
+ *
+ * @param {object} parameters Parameters
+ * @param {object} parameters.tree Project tree as generated by the
+ *   [@ui5/project.normalizer]{@link module:@ui5/project.normalizer}
+ * @param {Array.<string>} parameters.includeDependency The dependencies to be considered in 'includedDependencies'; the
+ *   "*" character can be used as wildcard for all dependencies and is an alias for the CLI option "--all"
+ * @param {Array.<string>} parameters.includeDependencyRegExp Strings which are interpreted as regular expressions
+ *   to describe the selection of dependencies to be considered in 'includedDependencies'
+ * @param {Array.<string>} parameters.includeDependencyTree The dependencies to be considered in 'includedDependencies';
+ *   transitive dependencies are also appended
+ * @param {Array.<string>} parameters.excludeDependency The dependencies to be considered in 'excludedDependencies'
+ * @param {Array.<string>} parameters.excludeDependencyRegExp Strings which are interpreted as regular expressions
+ *   to describe the selection of dependencies to be considered in 'excludedDependencies'
+ * @param {Array.<string>} parameters.excludeDependencyTree The dependencies to be considered in 'excludedDependencies';
+ *   transitive dependencies are also appended
+ * @param {Array.<string>} parameters.defaultIncludeDependency Same as 'includeDependency' parameter; used for build
+ *   settings
+ * @param {Array.<string>} parameters.defaultIncludeDependencyRegExp Same as 'includeDependencyRegExp' parameter; used
+ *   for build settings
+ * @param {Array.<string>} parameters.defaultIncludeDependencyTree Same as 'includeDependencyTree' parameter; used for
+ *   build settings
+ * @returns {object<string, string[]>} An object containing the 'includedDependencies' and 'excludedDependencies'
```

Comment:
Generics (<TypeParam, TypeParam...>), can only be used with the upper case Object, not with the primitive "object".

Reply:
Done

Reply:
I just had to revert to the lowercase style because the UI5 Tooling eslint rules doesn't allow the uppercase notation.

Reply:
Then please validate it locally: `npm run jsdoc`

Reply:
Sorry, then the rule is wrong, disable it.

Reply:
I've updated my last comment with a smiley to indicate that I was not 100% serious, only 95%.

Reply:
> I've updated my last comment with a smiley to indicate that I was not 100% serious, only 95%.

Thank you! It left me quite puzzled 😵‍💫

In the UI5 Tooling JSDoc it seems that both work. The difference in the HTML is simply:
> object.<string, Array.<string>>

vs
> Object.<string, Array.<string>>

In this case you're right that we indeed return an `Object` but it's also an `object`. From the [rule](https://github.com/gajus/eslint-plugin-jsdoc/blob/master/.README/rules/check-types.md):
> To make things more confusing, there are also object literals and object Objects. But object literals are still static Objects and object Objects are instantiated Objects. So an object primitive is still an object Object.

Reply:
Rule documentation IMO is only a second level of truth. I refer to https://jsdoc.app/tags-type.html and https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System . 

But you're free to ignore me. I didn't "request changes", I only commented.

Reply:
> But you're free to ignore me. 

Will do 😉 

Let's keep it as-is for now. IMHO the documentation is good enough 🥉 . Changing the ESLint configuration across all projects is just too much work to make this (private) documentation perfect.

Related discussion in eslint-plugin-jsdoc: https://github.com/gajus/eslint-plugin-jsdoc/issues/709

Reply:
Or add a single line disable. I'll leave that to @larskissel